packages/controllers/src/snaps/SnapController.test.ts

```diff
@@ -835,6 +836,71 @@ describe('SnapController', () => {
     await eventSubscriptionPromise;
   });
 
+  it('should error on invalid semver range during installSnaps', async () => {
+    const controller = getSnapController();
+    const origin = 'foo.com';
+
+    const result = await controller.installSnaps(origin, {
+      [FAKE_SNAP_ID]: { version: 'foo' },
+    });
+
+    expect(result).toMatchObject({
+      [FAKE_SNAP_ID]: { error: expect.any(EthereumRpcError) },
+    });
+  });
+
+  it('should reuse an already installed Snap if satisfies requestes semver range', async () => {
+    const messenger = getSnapControllerMessenger();
+    const controller = getSnapController(
+      getSnapControllerOptions({ messenger }),
+    );
+    const origin = 'foor.com';
+
+    const snap = await controller.add({
+      id: FAKE_SNAP_ID,
+      manifest: getSnapManifest(),
+      sourceCode: FAKE_SNAP_SOURCE_CODE,
+    });
+
+    const addSpy = jest.spyOn(controller as any, 'add');
+    const authorizeSpy = jest.spyOn(controller as any, 'authorize');
+    const messengerCallMock = jest
+      .spyOn(messenger, 'call')
+      .mockImplementationOnce(() => true);
+
+    await controller.installSnaps(origin, {
+      [FAKE_SNAP_ID]: { version: '>0.9.0 <1.1.0' },
+    });
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const newSnap = controller.get(FAKE_SNAP_ID)!;
+
+    // Notice usage of toBe - we're checking if it's actually the same object, not an equal one
```

Comment:
Hmm, why is that important in this context?

Reply:
To ensure that we don't actually manipulate the snap in any way